teps/000X-trigger-crd.md

```diff
@@ -0,0 +1,351 @@
+kind: EventListener
+metadata:
+  name: my-repo-trigger
+spec:
+  serviceAccountName: "blah"
+  boundnamespace:
+      - foo-ns
+      - bar-ns
+```
+
+### Path based EventListener
+Inside EventListener, we will deduce Trigger to execute based on the path of the request URL.
+To refer to trigger resource foo in namespace bar, we would url ```/bar/foo```. Instead of 
+EventListener iterating through every Triggers that can be serve by EventListener, EventListener
+will directly process the Trigger refer by URL.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this TEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Trigger CRD Implmentation
+Trigger CRD will be implemented. Trigger CRD spec will contain serviceaccount, triggerbinding, 
+triggertemplate and interceptor. ServiceAccount will be optional. If not defined, ServiceAccount
+of EventListener will be used.
+
+#### EventListener Selector Implementation
+Selector based EventListener implementation will be introduced. All namespace matching the 
+boundnamespaces in EventListener will be served by EventListener.
+
+#### Path Base EventListener Implementation
+Path base EventListener will be implemented.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+
+### User Experience (optional)
+
+<!--
+Consideration about the user experience. Depending on the area of change,
+users may be task and pipeline editors, they may trigger task and pipeline
+runs or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+
+### Performance (optional)
+Path base EventListener will improve the performance of EventListener by directly targetting
+the trigger definition we want to target instead of needlessly processing trigger that we don't.
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
```

Comment:
i think there's a risk to performance here as well - if one eventlistener is handling requests for all of your namespaces, and each namespace is a different customer (is it? i'm not sure but might be something you can flesh out in use cases), one poorly behaved customer/namespace could dominate an eventlistener and affect other namespaces, which wouldn't be the case if each namespace had their own eventlistener

Reply:
Yes, I added that and mitigation for that also.